pkgs/development/python-modules/llm/default.nix

```diff
@@ -0,0 +1,115 @@
+{
+  buildPythonApplication,
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  makeWrapper,
+  pytestCheckHook,
+  python3,
+  pythonOlder,
+  ruff,
+  setuptools,
+}: let
+
+  llm = buildPythonPackage rec {
+    pname = "llm";
+    version = "0.12";
+    pyproject = true;
+
+    disabled = pythonOlder "3.8";
+
+    src = fetchFromGitHub {
+      owner = "simonw";
+      repo = pname;
+      rev = "refs/tags/${version}";
+      hash = "sha256-aCqdw2co/cXrBwVY/k/aSLl3C22nlH5LvU2yir1/NnQ=";
+    };
+
+    nativeBuildInputs = [
+      setuptools
+    ];
+
+    propagatedBuildInputs = with python3.pkgs; [
+      click-default-group
+      numpy
+      openai
+      pip
```

Comment:
pip at runtime? You are sure?

Reply:
Pip at runtime was used mostly because there are two commands inside the cli (`install` and `uninstall`) that were calling `pip` to install external plugins.

On a second thought it doesn't make sense in the context of nix as any plugin will be installed using the provided `withPlugins` clause. So, I have introduced a patch that disables both commands and just print to the terminal instructions on how to proceed with `withPlugins`.